fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/BaseWhereAdapter.java

```diff
@@ -0,0 +1,1137 @@
+     * @return
+     */
+    public T not(ExpNode expr) {
+        predicateParser.not();
+        predicateParser.addToken(expr);
+        return getThis();
+    }
+
+    /**
+     * Add * to the expression
+     * @return
+     */
+    public T mult() {
+        predicateParser.mult();
+        return getThis();
+    }
+
+    /**
+     * Add + to the expression
+     * @return
+     */
+    public T add() {
+        predicateParser.add();
+        return getThis();
+    }
+
+    /**
+     * Add - to the expression
+     * @return
+     */
+    public T sub() {
+        predicateParser.sub();
+        return getThis();
+    }
+
+    /**
+     * Add / to the expression
+     * @return
+     */
+    public T div() {
+        predicateParser.div();
+        return getThis();
+    }
+
+    /**
+     * Add a COALESCE(c1, c2, ...) function. These should actually be column
+     * expressions, but for now we only need to model them as table.col values
+     * @param columnRefs
+     * @return
+     */
+    public T coalesce(ColumnRef... columnRefs) {
+        predicateParser.addToken(new CoalesceExpNode(columnRefs));
+        return getThis();
+    }
+
+    /**
+     * Add SIN(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T sin(ExpNode arg) {
+        predicateParser.addToken(new SinExpNode(arg));
+        return getThis();
+    }
+
+    /**
+     * Add SIN(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T sin(ColumnRef arg) {
+        predicateParser.addToken(new SinExpNode(new ColumnExpNode(arg.getRef())));
+        return getThis();
+    }
+
+    /**
+     * Add COS(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T cos(ExpNode arg) {
+        predicateParser.addToken(new CosExpNode(arg));
+        return getThis();
+    }
+
+    /**
+     * Add COS(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T cos(ColumnRef arg) {
+        predicateParser.addToken(new CosExpNode(new ColumnExpNode(arg.getRef())));
+        return getThis();
+    }
+
+    /**
+     * Add ACOS(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T acos(ExpNode arg) {
+        predicateParser.addToken(new ACosExpNode(arg));
+        return getThis();
+    }
+
+    /**
+     * Add ACOS(arg) to the expression
+     * @param arg
+     * @return
+     */
+    public T acos(ColumnRef arg) {
+        predicateParser.addToken(new ACosExpNode(new ColumnExpNode(arg.getRef())));
+        return getThis();
+    }
+
+    /**
+     * Add an IS NOT NULL operator to the expression
+     * @return
+     */
+    public T isNotNull() {
+        predicateParser.isNotNull();
+        return getThis();
+    }
+
+
+    /**
+     * Add an IS NULL operator to the expression
+     * @return
+     */
+    public T isNull() {
+        predicateParser.isNull();
+        return getThis();
+    }
+
+    /**
+     * Add a left paren `(` to the expression
```

Comment:
```suggestion
     * Add a left paren <code>(</code> to the expression
```

mixing markdown and javadoc

Reply:
ok